crates/bevy_ui/src/flex/mod.rs

```diff
@@ -244,7 +254,8 @@ pub fn flex_node_system(
         flex_surface.upsert_leaf(entity, style, *calculated_size, scale_factor);
     }
 
-    // TODO: handle removed nodes
+    // clean up removed nodes
+    flex_surface.remove_entities(removed_nodes.iter());
```

Comment:
```suggestion
    flex_surface.remove_entities(removed_nodes);
```

Reply:
This doesn't work because `RemovedComponent` doesn't implement `IntoInterator`

Reply:
It would be nice if it did.
Do you mind adding the following snippet [here](https://github.com/bevyengine/bevy/tree/main/crates/bevy_ecs/src/system/system_param.rs#L782)?
```rust
impl<'a, T: Component> IntoIterator for &'a RemovedComponents<'a, T> {
    type Item = Entity;
    type IntoIter = std::iter::Cloned<std::slice::Iter<'a, Entity>>;
    
    fn into_iter(self) -> Self::IntoIter {
        self.iter()
    }
}
```
And then change this line to
```suggestion
    flex_surface.remove_entities(&removed_nodes);
```